docusaurus/docs/tutorials/build-an-app-plugin.md

```diff
@@ -0,0 +1,202 @@
+- NodeJS 12.x or later.
+- yarn
+
+## Create a new plugin
+
+<CreatePlugin pluginType="app" />
+
+## Anatomy of a plugin
+
+<PluginAnatomy />
+
+## App plugins
+
+App plugins let you bundle resources such as dashboards, panels, and data sources into a single plugin.
+
+Any resource you want to include needs to be added to the `includes` property in the `plugin.json` file. To add a resource to your app plugin, you need to include it to the `plugin.json`.
+
+Plugins that are included in an app plugin are available like any other plugin.
+
+Dashboards and pages can be added to the app menu by setting `addToNav` to `true`.
+
+By setting `"defaultNav": true`, users can navigate to the dashboard by clicking the app icon in the side menu.
+
+### Add a custom page
+
+App plugins let you extend the Grafana user interface through the use of _custom pages_.
+
+Any requests sent to `/a/<plugin-id>`, e.g. `/a/myorgid-simple-app/`, are routed to the _root page_ of the app plugin. The root page is a React component that returns the content for a given route.
+
+While you're free to implement your own routing, in this tutorial you'll use a tab-based navigation page that you can use by calling `onNavChange`.
+
+Let's add a tab for managing server instances.
+
+1. In the `src/pages` directory, add a new file called `Instances.tsx`. This component contains the content for the new tab.
+
+   ```ts
+   import { AppRootProps } from '@grafana/data';
+   import React, { FC } from 'react';
+
+   export const Instances: FC<AppRootProps> = ({ query, path, meta }) => {
+     return <p>Hello</p>;
+   };
+   ```
+
+1. Register the page by adding it to the `pages` array in `src/pages/index.ts`:
+
+   **index.ts**
+
+   ```ts
+   import { Instances } from './Instances';
+   ```
+
+   ```ts
+   {
+     component: Instances,
+     icon: 'file-alt',
+     id: 'instances',
+     text: 'Instances',
+   }
+   ```
+
+1. Add the page to the app menu by including it in `plugin.json`. This will be the main view of the app, so we'll set `defaultNav` to let users quickly get to it by clicking the app icon in the side menu.
+
+   **plugin.json**
+
+   ```json
+   "includes": [
+     {
+       "type": "page",
+       "name": "Instances",
+       "path": "/a/myorgid-simple-app?tab=instances",
+       "role": "Viewer",
+       "addToNav": true,
+       "defaultNav": true
+     }
+   ]
+   ```
+
+> **Note:** While `page` includes typically reference pages created by the app, you can set `path` to any URL, internal or external. Try setting `path` to `https://grafana.com`.
+
+### Configure the app
+
+Let's add a new configuration page where users are able to configure default zone and regions for any instances they create.
+
+- In `module.ts`, add new configuration page using the `addConfigPage` method. `body` is the React component that renders the page content.
+
+  **module.ts**
+
+  ```ts
+  .addConfigPage({
+    title: 'Defaults',
+    icon: 'fa fa-info',
+    body: DefaultsConfigPage,
```

Comment:
Maybe call this `AppConfig`? Updating app plugin configurations isn't as straight forward as you'd want it to be...Would be good to also show an example of how to configure default time zones and regions in the app config component and call the backend to save the updates.